packages/nx/src/utils/ignore-patterns.ts

```diff
@@ -0,0 +1,192 @@
+import { Dirent, existsSync, readdirSync } from 'fs';
+import { readFileSync } from 'fs-extra';
+import { readdir } from 'fs/promises';
+import ignore, { Ignore } from 'ignore';
+import { join, relative } from 'path';
+import { logger } from './logger';
+import { normalizePath } from './path';
+import { workspaceRoot } from './workspace-root';
+
+const ALWAYS_IGNORE = [join('node_modules'), join('.git')];
+
+export const locatedIgnoreFiles = new Set<string>();
+
+export interface GetIgnoredGlobsOptions {
+  ig?: Ignore;
+  nxIgnoreOnly?: boolean;
+}
+
+function getGlobOptions(
+  provided?: GetIgnoredGlobsOptions
+): Required<GetIgnoredGlobsOptions> {
+  return {
+    ig: provided?.ig ?? ignore(),
```

Comment:
We can probably get rid of this if we have a private util to create an ignore test function